[Sum Hung Yee] iP

sumhungyee opened the pull request:
# DukeyGPT0.01

> "I fear the day when people post memes on the internet with quotes I've never said." — Albert Einstein

DukeyGPT0.01 frees your mind of having to remember things you need to do. It,
- Reminds you of ChatGPT!
- **Reminds you of ChatGPT!**
- **_Reminds you of ChatGPT!!!!!_**
- Is super easy to use!
- Is ~~fast~~ SUPER FAST!

It keeps tracks of tasks, like these!
- [x] Use DukeyGPT0.01
- [x] **Use DukeyGPT0.01**
- [ ] Deadline **_USE DukeyGPT0.01 OR ELSE_** (by: 2022-02-02 00:00)

1. For legal reasons the above is a joke 🤡. [But here is the download link.](https://github.com/sumhungyee/ip/releases/download/A-Jar/Duke.jar)
2. Download it. 🤖 
3. [Take a break. Watch this 38-min footage of a walking simulator around ancient Egypt.](https://www.youtube.com/watch?v=XsEBbVoSkJY)
4. I like purple. `#8900ff`


``` r
# this is some R code 
fun <- function(x) {
    return(ifelse(x %% 2, x, x ^ 2))
}
```

``` python
# here is some python code
def fun(x):
    return x if x % 2 else x ** 2
```

Unfortunately, I'm not very familiar with Java :(

Commit:
Add Gradle support

sumhungyee committed:
Added level 1

sumhungyee committed:
Added level 2

sumhungyee committed:
Added level 3

sumhungyee committed:
Added level 4

sumhungyee committed:
A-TextUiTesting

sumhungyee committed:
level-5

sumhungyee committed:
level-6

sumhungyee committed:
level-6

sumhungyee committed:
modofied level 6

sumhungyee committed:
edit main class

sumhungyee committed:
add level-7

sumhungyee committed:
add level-8

sumhungyee committed:
resolve merge manually

sumhungyee committed:
edit formatting of localdatetime

sumhungyee committed:
edit level-7 printing

sumhungyee committed:
add header comments

sumhungyee committed:
add A-MoreOOP

sumhungyee committed:
add A-MoreOOP

sumhungyee committed:
add A-MoreOOP

sumhungyee committed:
add A-Packages

sumhungyee committed:
add A-Gradle

sumhungyee committed:
add tests

sumhungyee committed:
add tests

sumhungyee committed:
add A-JUnit

sumhungyee committed:
add A-JUnit

sumhungyee committed:
add

sumhungyee committed:
add A-JavaDoc

sumhungyee committed:
add A-JavaDoc

sumhungyee committed:
add A-JavaDoc

sumhungyee committed:
add level-9

sumhungyee committed:
add A-JavaDoc

sumhungyee committed:
add Level-9

sumhungyee committed:
merge Level-9 manually

sumhungyee committed:
edit parser to accomodate for events

sumhungyee committed:
add more exception handling

sumhungyee committed:
merge add-gradle-support

sumhungyee committed:
add more exceptions

sumhungyee committed:
add checkstyle support

sumhungyee committed:
add checkstyle support

sumhungyee committed:
add checkstyle support

sumhungyee committed:
add error messages

sumhungyee committed:
add more JavaDoc comments

sumhungyee committed:
add more JavaDoc comments

sumhungyee committed:
add more JavaDoc comments

sumhungyee committed:
add more changes for checkstyle

sumhungyee committed:
add A-CheckStyle

sumhungyee committed:
add level-10

sumhungyee committed:
resolve merges

sumhungyee committed:
modified level 10

sumhungyee committed:
modified level 10

sumhungyee committed:
modified level 10

Mehvin commented on src/main/java/duke/duke/Duke.java:
```diff
@@ -0,0 +1,106 @@
+        }
+
+    }
+
+
+
+
+
+
+
```
Irrelevant whitespaces can be removed to make code neater

Mehvin commented on src/main/java/duke/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+            c = new TodoCommand(new Todos(description));
+            break;
+        case "event":
+            if (line.length == 1) {
+                throw new NoArgsException("deadline");
+            } else if (joined.split("/").length != 3) {
+                throw new InvalidException();
+            }
+
+            queries = this.queries(line, Events.KEYWORDS);
+            Events event = new Events(queries);
+            c = new EventCommand(event);
+            break;
+        case "deadline":
+            if (line.length == 1) {
+                throw new NoArgsException("deadline");
+            } else if (joined.split("/").length != 2) {
+                throw new InvalidException();
+            }
+
+            queries = this.queries(line, Deadlines.KEYWORDS);
+            c = new DeadLineCommand(new Deadlines(queries));
+            break;
+
+        case "bye":
+            c = new ByeCommand();
+            break;
+
+        case "find":
+            if (line.length == 1) {
+                throw new NoArgsException("deadline");
+            }
+            String query = this.queries(line, List.<String>of()).get(0);
+            c = new FindCommand(query);
+            break;
+
+        default:
+            throw new UnrecognisableException();
+        }
+        return c;
```
Great work on the formatting of switch statement here 👍

Mehvin commented on src/main/java/duke/duke/Ui.java:
```diff
@@ -0,0 +1,90 @@
+    private final Scanner scanner;
+
+    Ui() {
+        this.logo = "....................*******************************...................\n"
```
Cool logo 🤖

Mehvin commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,101 @@
+     */
+    public TaskList loadFile() throws LoadException {
+
+        ArrayList<Task> loadData;
```
Perhaps a more intuitive variable name can be used for `loadData`
```suggestion
        ArrayList<Task> loadedDataList;
```

Mehvin reviewed (commented):
Overall very little mistakes were made, I liked that most of the JavaDocs are in place too! 👍

sumhungyee committed:
add checkstyle changes

sumhungyee committed:
Add assertions to the "duke", "storage" and "commands" packages

There are some classes such as Parser and TaskList that require
assertions to perform checks on variables.

For instance, when we edit the TaskList, such as when deleting,  we must ensure that a TaskList
has a size > 0. This ensures that we do not overdelete.

In Parser, we want to verify that the "bye" command must have a
.isBye() that returns true,  such that the while loop in Duke can be
closed.

sumhungyee committed:
Remove additional whitespaces in the Parser

There are too many newlines in between lines of code. Sometimes,
these newlines are not consistent, such as in the Parser class, where
1 or 2 or 0 empty lines are found between cases for the switch
  statement. To improve
readability, these newlines are removed.

sumhungyee committed:
Merge pull request #1 from sumhungyee/branch-A-Assertions

Add assertions to the "duke", "storage" and "commands" packages

sumhungyee committed:
Merge pull request #2 from sumhungyee/branch-A-CodeQuality

Remove additional whitespaces in the Parser

sumhungyee committed:
change variable name from loadData to taskList

sumhungyee committed:
fix syntax error by adding ':'

sumhungyee committed:
add exception handling to taskList.remove()

sumhungyee committed:
Add streams to the dumpFile method in Storage

As part of the A-Streams level, using streams replaces the previous
implementation, which uses for loops.

For loops make the code extremely unreadable and messy, and they appear
inefficient.
Using streams makes the code more readable and efficient as compared to for loops.

sumhungyee committed:
Merge pull request #3 from sumhungyee/branch-A-Streams

Add streams to the dumpFile method in Storage

sumhungyee committed:
add A-CI

sumhungyee committed:
Merge branch 'master' of https://github.com/sumhungyee/ip

sumhungyee committed:
add A-Undo

sumhungyee committed:
add C-Undo

sumhungyee committed:
edit tasklist getTaskStrings(()

sumhungyee committed:
add  A-BetterGui

sumhungyee committed:
edit gradle.yml to remove redirection tests

sumhungyee committed:
add A-Personality

sumhungyee committed:
add acknowledgements to DialogBox.fxml

sumhungyee committed:
add Ui.png

sumhungyee committed:
Update Readme.md

sumhungyee committed:
Update README.md

sumhungyee committed:
Update README.md

sumhungyee committed:
Update logo in Readme.md

sumhungyee committed:
add fxml

sumhungyee committed:
replace fxml files with non-template fxml files, which were used for debugging purposes

sumhungyee committed:
remove bugs in TaskList that does not inform the user of an empty list and add functionality to undo to accept 1 argument

sumhungyee committed:
edit user guide

sumhungyee committed:
remove bugs

sumhungyee committed:
debug

sumhungyee committed:
debug

sumhungyee committed:
Merge branch 'master' of https://github.com/sumhungyee/ip

sumhungyee committed:
edit DialogBox

sumhungyee committed:
debug all files

sumhungyee committed:
resolve bugs

sumhungyee committed:
resolve bugs

sumhungyee committed:
remove bugs on saving file

sumhungyee committed:
add A-UserGuide

sumhungyee committed:
edit image in website

sumhungyee committed:
edit user guide
